packages/back-end/src/events/base-types.ts

```diff
@@ -29,13 +30,15 @@ export const notificationEventResources = [
   "user",
   "webhook",
 ] as const;
+
 export type NotificationEventResource = typeof notificationEventResources[number];
 
 /**
  * Event Notification payload
  */
 export type NotificationEventPayload<
-  EventName extends NotificationEventName,
+  EventName extends NotificationEventName &
+    (ResourceType extends string ? `${ResourceType}.${string}` : unknown),
```

Comment:
This makes sure that `EventName` is always of the form: `${ResourceType}.<string>`